### ISSUE TITLE
When user tabs through the modal dialog for the first time, focus goes out of the dialog to the Show Modal button

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4598]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 2.4.3 Focus Order

**Test Environment:**
- Test URL: https://clarity.design/documentation/modals
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 19.3.1

**Steps to reproduce:**
1. Open the test URL.
2. Move focus onto to the first Show Modal button present under Angular modals.
3. Activate the button by pressing the enter or space key and perform tab navigation.

**Expected result:**
Focus should be trapped within the dialog.

**Actual result:**
For the first time with forward tab navigation focus goes out of the dialog and goes to the Show Modal button. If the user presses the tab key again, the focus comes back to the dialog and stays within the dialog window tab key navigation.

**Note:** This issue is not observed with arrow key navigation.

**Reporter Email:** gkhadke@vmware.com

**Comment by gnomeontherun** [comment]
Hi there, this is an automated message.

In an effort to dedupe and track accessibility reports consistently, this issue is being closed and tracked internally with our Accessibility team. It will still be triaged and addressed as part of ongoing work.
